src/utils/byte_utils.ts

```diff
@@ -26,7 +26,7 @@ export type ByteUtils = {
   /** Create a Uint8Array containing utf8 code units from a string */
   fromUTF8: (text: string) => Uint8Array;
   /** Create a string from utf8 code units */
-  toUTF8: (buffer: Uint8Array) => string;
+  toUTF8: (buffer: Uint8Array, start?: number, end?: number) => string;
```

Comment:
Can we make start and end required? I think there's only a few usages that would need updating.